packages/memsocket/default.nix

```diff
@@ -0,0 +1,49 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  stdenv,
+  lib,
+  debug,
+  vms,
+  gcc,
+  gnumake,
+  fetchFromGitHub,
+  ...
+}:
+stdenv.mkDerivation {
+  name = "memsocket";
+
+  src = fetchFromGitHub {
+    owner = "tiiuae";
+    repo = "shmsockproxy";
+    rev = "4eef1a7e1f6993a9ba0ef1aa066145c410e2eec6";
+    sha256 = "sha256-uaKBtb+bD42RK/Z96dhMrGxNXg+ZV/6iW52ouNQ+QSY=";
+  };
+
+  nativeBuildInputs = [gcc gnumake];
+
+  CFLAGS =
+    "-O2 -DVM_COUNT="
+    + (toString vms)
+    + (
+      if debug
+      then " -DDEBUG_ON"
+      else ""
+    );
+  prePatch = ''
+    cd app
```

Comment:
If app not in repo root, but in `app/` subdirectory -- use `sourceRoot="app/"` statement, not chdir during prePatch phase

Reply:
Done. Surprisingly, the `sourceRoot` must be headed with `source`, i.e.: `sourceRoot="source/app/"`.